src/lib/menu/menu-trigger.ts

```diff
@@ -216,23 +216,34 @@ export class MdMenuTrigger implements AfterViewInit, OnDestroy {
 
   /** Closes the menu. */
   closeMenu(): void {
+    this._closeMenu(true);
+  }
+
+  /** Focuses the menu trigger. */
+  focus() {
+    this._element.nativeElement.focus();
+  }
+
+  /**
+   * Gives the option of setting emitEvent to false to avoid emitting an event
+   * when an event will already be emitted.
+   */
+  private _closeMenu(emitEvent: boolean) {
```

Comment:
Rather than having the `emitEvent` parameter, you can remove the `this.menu.close.emit` call from here and move it only to `closeMenu`. Also change the doc string to something along the lines of "Closes the menu and does the necessary cleanup".

Reply:
Done. Should I only emit the event when the menu is already open, though?